regimes/se/tax_identities.go

```diff
@@ -0,0 +1,79 @@
+package se
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/invopop/gobl/cbc"
+	"github.com/invopop/gobl/tax"
+	"github.com/invopop/validation"
+)
+
+const (
+	taxCodeCountryPrefix = "SE"
+	taxCodeLength        = 14
+	taxCodeCheckDigit    = "01"
+)
+
+var (
+	// ErrInvalidTaxIDLength is returned when the tax ID is not the correct length.
+	ErrInvalidTaxIDLength = errors.New("invalid length")
+	// ErrInvalidTaxIDCountryPrefix is returned when the tax ID does not start with "SE".
+	ErrInvalidTaxIDCountryPrefix = errors.New("invalid country prefix")
+	// ErrInvalidTaxIDCheckDigit is returned when the tax ID does not end with "01".
+	ErrInvalidTaxIDCheckDigit = errors.New("invalid check digit")
+	// ErrInvalidTaxIDCharacters is returned when the tax ID contains non-numeric characters.
+	ErrInvalidTaxIDCharacters = errors.New("invalid characters, expected numeric")
+)
+
+// normalizeTaxIdentity performs normalization specific to Swedish tax IDs,
+// ensuring the code is normalized and the country prefix is added if missing.
+func normalizeTaxIdentity(id *tax.Identity) {
+	tax.NormalizeIdentity(id)
+	// TO-DO: decide if this is necessary. If not, we may need to remove the check digit suffix.
+	// Re-add the SE prefix if missing.
+	if id.Code.String()[:2] != taxCodeCountryPrefix {
+		id.Code = cbc.Code(taxCodeCountryPrefix + id.Code.String())
+	}
+}
+
+// validateTaxIdentity performs validation specific to Swedish tax IDs.
+// Assumes the code has already been normalized.
+func validateTaxIdentity(tID *tax.Identity) error {
+	return validation.ValidateStruct(tID,
+		validation.Field(&tID.Code,
+			validation.By(validateTaxCode),
+		),
+	)
+}
+
+// PERF: validate this is indeed faster than a RegEx.
```

Comment:
Performance isn't a high concern in GOBL. There is already a large amount of complexity so a pre-compiled Regexp has negligible impact.

Reply:
Then the question is if that would be easier to understand than a few `if` statements. Since Swedish tax code validation is so simple, I'd say this approach makes it more maintainable, but open to feedback!

Reply:
If you remove the comment, no one will question it ;-)

Reply:
The voice in the back of my mind certainly will... But I can live with that 😆